examples/AdvancedOperations/AddGmailAd.php

```diff
@@ -0,0 +1,251 @@
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param int $adGroupId the ad group ID to add an ad to
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId
+    ) {
+        $mediaFiles = self::addMediaFiles($googleAdsClient, $customerId);
+        self::addGmailAd($googleAdsClient, $customerId, $adGroupId, $mediaFiles);
+    }
+
+    /**
+     * Adds the image files.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @return array the image file resource names
+     */
+    private static function addMediaFiles(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId
+    ) {
+        // Creates the logo image data.
+        $logoImageData = file_get_contents(self::LOGO_IMAGE_URL);
+
+        // Creates the logo image.
+        $logoMediaFile = new MediaFile([
+            'type' => MediaType::IMAGE,
+            'image' => new MediaImage(['data' => $logoImageData]),
+            'mime_type' => MimeType::IMAGE_PNG
+        ]);
+
+        // Creates the operation for the logo image.
+        $logoMediaFileOperation = new MediaFileOperation();
+        $logoMediaFileOperation->setCreate($logoMediaFile);
+
+        // Creates the marketing image data.
+        $marketingImageData = file_get_contents(self::MARKETING_IMAGE_URL);
+
+        // Creates the marketing image.
+        $marketingMediaFile = new MediaFile([
+            'type' => MediaType::IMAGE,
+            'image' => new MediaImage(['data' => $marketingImageData]),
+            'mime_type' => MimeType::IMAGE_JPEG
+        ]);
+
+        // Creates the operation for the marketing image.
+        $marketingMediaFileOperation = new MediaFileOperation();
+        $marketingMediaFileOperation->setCreate($marketingMediaFile);
+
+        // Issues a mutate request to add the media files.
+        $mediaFileServiceClient = $googleAdsClient->getMediaFileServiceClient();
+        $response = $mediaFileServiceClient->mutateMediaFiles(
+            $customerId,
+            [$logoMediaFileOperation, $marketingMediaFileOperation]
+        );
+
+        foreach ($response->getResults() as $addedMediaFile) {
+            /** @var MediaFile $addedMediaFile */
+            printf(
+                "Created media file with resource name '%s'.%s",
+                $addedMediaFile->getResourceName(),
+                PHP_EOL
+            );
+        }
+
+        // Returns the created media file resource names.
+        return [
+            'logoResourceName' => $response->getResults()[0]->getResourceName(),
+            'marketingImageResourceName' => $response->getResults()[1]->getResourceName()
+        ];
+    }
+
+    /**
+     * Adds the gmail ad.
```

Comment:
Gmail should be capitalized as it is in other comments.

Reply:
Done